src/components/menus/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import * as styles from './ContextMenu.scss';
+import { DotsIcon } from '../../icons/Icons';
+import { FunctionGeneric } from '../../../common/structures/Generics';
+import { Tooltip, TooltipProps } from '../../overlays/Tooltip/Tooltip';
+import { Button } from '../../buttons/Button/Button';
+import { TextButton } from '../../buttons/TextButton/TextButton';
+
+export interface IMenuItem {
+    color?: 'red' | 'none';
+    className?: string;
+    content?: React.ReactNode;
+    label?: string;
+    onClick?: FunctionGeneric;
+    enabled?: boolean;
+    type?: 'normal' | 'separator';
+}
+
+export interface IContextMenuProps extends Omit<TooltipProps, 'content'> {
+    /** className for the list items' container */
+    classNameList?: string;
+    /** className for an individual list item */
+    classNameListItem?: string;
+    /** array of menu items */
+    items: IMenuItem[];
+}
+
+const ContextMenu = (props: IContextMenuProps) => {
+    const { className, classNameList, classNameListItem, items, onShow, onHide, ...otherProps } = props;
+
+    const onClickItem = (event: React.MouseEvent<HTMLLIElement, MouseEvent>, item: IMenuItem) => {
+        item.onClick?.call(null);
+        event.stopPropagation();
+    };
+
+    const [isShowing, setIsShowing] = React.useState(false);
+
+    const content = (
+        <ul className={classnames(styles.ContextMenu_Items, 'ContextMenu_Items', classNameList)}>
+            {items
+                .filter((item: IMenuItem) => Object.keys(item).length !== 0)
```

Comment:
Just a thought in case there are simpler use cases in the future, but instead of chaining filter and map here I think you could achieve the same with one reduce to save some iterations. It could get tricky with the types though and I like this for its readability.